### PR TITLE
Adds Remove Song Functionality

### PR DESCRIPTION
**Pull request by MPierre9**
Improved removed song functionality to work with latest merge #174  and #175 and addresses issue #147 .

## Example

`ap1.removeSong(0);` removes first song
`ap1.removeSong(1);` removes second song
etc.

aplayerremovesongv2

Had to close first PR because of rebasing issue.

### CODE REVIEW

**Comment by MPierre9** [comment]
@DIYgod Thoughts or changes that have to be made?

File: src/APlayer.js [hunk 1]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
     }
 
+      /**
+    * Remove song from playlist 
+    */
+    removeSong(indexOfSong) {
+        
+               if (this.option.music[indexOfSong] != null) { // Check if song exists 
+                   const list = this.element.getElementsByClassName('aplayer-list')[0];
+                   var oList = list.firstElementChild; // OL tag
```
Comment (DIYgod) [inline src/APlayer.js]: `list.getElementsByTagName('ol')[0]` is more straightforward

File: src/APlayer.js [hunk 2]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+                   var liList = []; // Holds the index of the LI tags
```
Comment (DIYgod) [inline src/APlayer.js]: Why do this, `oList.getElementsByTagName('li')` is more straightforward

File: src/APlayer.js [hunk 3]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+                   if (this.option.music[indexOfSong + 1] != null || this.option.music[indexOfSong - 1] != null) {
+                       if (indexOfSong == this.playIndex) {
+        
+                           if (this.option.music[indexOfSong + 1] != null) { // Play next song if it exists. If not paused.
+                               this.setMusic(indexOfSong + 1);
+                           } else if (this.option.music[indexOfSong + 1] == null) { // Play previous song if it exists. If not paused.
+                               this.setMusic(indexOfSong - 1);
+                           }
+             
+                           this.playIndex = this.playIndex - 1
```
Comment (DIYgod) [inline src/APlayer.js]: playIndex have set in `setMusic`, it is unnecessary to set it again

File: src/APlayer.js [hunk 4]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+        
+                       } else {
+                           if (indexOfSong < this.playIndex) {
+                               this.playIndex = this.playIndex - 1;   
+                           }
+                       }
+        
+                       if (oList.childNodes[liList[indexOfSong + 1]] == null) {
```
Comment (DIYgod) [inline src/APlayer.js]: `oList.childNodes[liList[indexOfSong + 1]` -> `liList[indexOfSong + 1]]`

File: src/APlayer.js [hunk 5]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+                           var targetSong = oList.childNodes[liList[indexOfSong - 1]];
+                           targetSong.childNodes[3].textContent = indexOfSong;
```
Comment (DIYgod) [inline src/APlayer.js]: `targetSong.childNodes[3]` ->`targetSong.getElementsByClassName('aplayer-list-index')`

File: src/APlayer.js [hunk 6]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+                               }
+                           }
+                       }
+                       this.option.music.splice(indexOfSong, 1); // Delete song from music array
+                       this.audios.splice(indexOfSong, 1); // Delete song from audios array (Has to be)
+                       oList.childNodes[liList[indexOfSong]].remove();
+        
+                       if (this.option.music[0] != null && this.option.music[1] == null) {
+                           this.multiple = false;
+                           this.element.classList.remove('aplayer-withlist');
```
Comment (DIYgod) [inline src/APlayer.js]: `remove` is not supported in IE, use `removeChild` instead

File: src/APlayer.js [hunk 7]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+                       }
+                   }
+        
+                   list.style.height = "";
```
Comment (DIYgod) [inline src/APlayer.js]: `list.style.height` is needed by hiding/showing list animation, it should not be empty

File: src/APlayer.js [hunk 8]
```diff
@@ -846,6 +846,66 @@ class APlayer {
         }
     }
 
+      /**
```
Comment (DIYgod) [inline src/APlayer.js]: incorrect indent

File: src/APlayer.js [hunk 9]
```diff
@@ -846,6 +846,66 @@ class APlayer {
...
+    * Remove song from playlist 
+    */
+    removeSong(indexOfSong) {
+        
+               if (this.option.music[indexOfSong] != null) { // Check if song exists
```
Comment (DIYgod) [inline src/APlayer.js]: incorrect indent

File: src/APlayer.js (hunk 6 above)
Comment (MPierre9) [reply src/APlayer.js]: Tried using removeChild and resulted in error. Shouldn't it be fine if APlayer.js contains remove() already? Example `this.lrcContents.getElementsByClassName('aplayer-lrc-current')[0].classList.remove('aplayer-lrc-current');
`

File: src/APlayer.js (hunk 7 above)
Comment (MPierre9) [reply src/APlayer.js]: Ok I changed code to: 
`var listHeight = parseInt(list.style.height, 10);`
`list.style.height = (listHeight - 33) + "px";`
Animation runs when song is removed.

File: src/APlayer.js (hunk 2 above)
Comment (MPierre9) [reply src/APlayer.js]: That's a lot easier, thanks. Change made.

**Pull request merged**